2022/python/day4.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Dec 26 11:30:48 2022
+
+@author: raegoering
+
+--- Day 4: Camp Cleanup ---
+
+"""
+
+def range2list(num_range):
+    """
+
+    Parameters
+    ----------
+    num_range : str describing a range ex. "2-4"
+
+    Returns
+    -------
+    num_list : list of integers in that range
+
+        This function converts a string describing a range to an integer list.
+    """
+    start = int(num_range.split(sep="-")[0])
+    end = int(num_range.split(sep="-")[1])
+    num_list = list(range(start, end + 1))
+    return num_list
+    
+def contained_within(num_list1, num_list2):
+    """
+
+    Parameters
+    ----------
+    num_list1 : list of integer chores of first elf
+    num_list2 : list of integer chores of second elf
+
+    Returns
+    -------
+    bool
+
+        This function determines if a chore list is contained within another
+
+    """
+    if num_list1[0] <= num_list2[0] and num_list1[-1] >= num_list2[-1]:
```

Comment:
If you convert the list1 and list2 to the builtin [set]() data structure, you can use the set methods to check these things.
```
s1 = set(num_list1)
s2 = set(num_list2)

if s2.issubset(s1) or s1.issubset(s2):
    return True
```

This is good for readability since someone else reading your code will probably also be familiar with sets